### ISSUE TITLE
Supporting Maven and Gradle Wrapper in JFrog Xray audit-gradle and audit-mvn

### ISSUE THREAD

**Issue by e88z4** [jfrog/jfrog-cli#1396]
**Is your feature request related to a problem? Please describe.**
I am trying to execute JFrog Cli with the command
jfrog audit-gradle

I got and error below:
[Error] exec: "gradle": executable file not found in $PATH

It seems the jfrog cli relies on gradle on the path which may be installed as a system wide gradle installation. 

**Describe the solution you'd like to see**
I prefer the jfrog cli for audit-gradle to support gradle wrapper. This can be extended to maven wrapper as well. Our environment doesn't have system wide installation of Gradle/Maven in the system $PATH.

**Comment by eyalbe4** [comment]
Thank you for sharing this requirement with us @e88z4!
We've added this task to our work backlog, and will update this issue once the work on this starts.

**Comment by yahavi** [comment]
@e88z4
Actually, using the Gradle wrapper in the `audit-gradle` command is already supported.
All you need to do is to create a minimal configuration file in:
`<workspace>/.jfrog/projects/gradle.yaml` 
or 
`<user-home>/.jfrog/projects/gradle.yaml`. 
The JFrog CLI generates this file by running one of the following commands:
```
# Option 1: Non interactive - local configuration
jf gradle-config --use-wrapper

# Option 2: Non interactive - global configuration
jf gradle-config --use-wrapepr --global

# Interactive
jf gradle-config
```

You can commit this file or create it every time in your CI pipeline.
If you wish, you also have the option to resolve the dependencies in the audit commands from Artifactory. See all options of the `gradle-config` command under [Running Gradle Builds](https://www.jfrog.com/confluence/display/CLI/CLI+for+JFrog+Artifactory#CLIforJFrogArtifactory-RunningGradleBuilds). See also [audit-gradle documentation](https://www.jfrog.com/confluence/display/CLI/CLI+for+JFrog+Xray#CLIforJFrogXray-AuditingGradleProjects) and [this](https://jfrog.com/blog/scanning-dependencies-in-your-sources-using-jfrog-cli-and-xray/) blog post.

Notwithstanding, using the Maven wrapper is not yet supported, but will be in the future.
Please let me know if that helped.

**Comment by e88z4** [comment]
Hi @yahavi , 

The Gradle wrapper instruction works. 

Can the ticket be open to fully support Maven wrapper?

**Comment by eyalbe4** [comment]
Definitely @e88z4.
We will use this issue and update it once we get to supporting the maven wrapper.
Thanks for feedback!

**Comment by ToniA** [comment]
The `jf gradle-config` should also support the `--uses-plugin` switch, to add `usePlugin: true` into the generated `.jfrog/projects/gradle.yaml` file.

This command also doesn't work with the jfrog-azure-devops-extension, when attempting to use the JfrogCliV2@1 task, as the task adds flags which are not supported by the `jf gradle-config` command. `jf gradle-config` also can't be easily ran from a generic script task, as it's not in the `$PATH`.

**Comment by ToniA** [comment]
Sorry, I mixed things up. Yes, `jf gradle-config` does support it. The problem is that there doesn't seem to be a way to run  `jf gradle-config` within an Azure DevOps JFrog task.

The JFrog Gradle task is just fine, but I'm interested in the JFrog Audit (for Gradle projects). Committing `.jfrog/projects/gradle.yaml` into the project is a clumsy solution. Right now I have a shell script task in Azure Devops to create this file with `cat` and here-documents.

**Comment by trevjonez** [comment]
I was able to workaround this by modifying the path to include the wrapper dist bin folder (before whatever other installation you have already if any) after running the wrapper once to initialize the install. Sub optimal for sure but unblocked my current task at least.

```bash
./gradlew -Dscan.tag.ScaScanPrep
echo "export PATH=\"$(find $HOME/.gradle/wrapper/dists -type d -regex ".*/bin" | head -1):\$PATH\"" >> $BASH_ENV
```

`$BASH_ENV` being a path to a script that CircleCi will run at the start of each run step so its useful for holding onto vars/functions/etc.

**Comment by coopernetes** [comment]
FYI, not sure if this is a separate issue but `jf` will routinely ignore the maven wrapper setting even when set on the CLI as well as in config. Logs from a GitHub Actions workflow below:

```
  cp .jfrog/projects/maven.yaml $HOME/.jfrog/projects/
  export PATH=$HOME/.local/bin:$PATH
  jf mvn-config --global --use-wrapper
  jf audit --mvn \
    --user $JFROG_USERNAME \
    --password $JFROG_PASSWORD \
    --url https://$JFROG_URL/xray \
    --watches $WATCHES \
    --format table \
    --extended-table
env:
    JAVA_HOME: /home/runner/_work/_tool/Java_Temurin-Hotspot_jdk/17.0.11-9/x64
    JFROG_USERNAME: ***
    JFROG_PASSWORD: ***
    JFROG_CLI_LOG_LEVEL: DEBUG
16:21:12 [Debug] JFrog CLI version: 2.58.2
16:21:12 [Debug] OS/Arch: linux/amd64
16:21:16 [Info] Running SCA scan for maven vulnerable dependencies in /home/runner/_work/artifactory/artifactory/tests/e2e/maven directory...
16:21:16 [Info] Calculating Maven dependencies...
16:21:16 [Debug] Using resolver config from /home/runner/_work/artifactory/artifactory/tests/e2e/maven/.jfrog/projects/maven.yaml
16:21:16 [Debug] Preparing to read the config file /home/runner/_work/artifactory/artifactory/tests/e2e/maven/.jfrog/projects/maven.yaml
16:21:16 [Debug] Analytics metrics are disabled, skipping sending update event request to XSC
16:21:16 [Debug] Sending an error report to JFrog analytics...
16:21:16 [Debug] Sending HTTP GET request to: https://<redacted>/xsc/api/v1/system/version
16:21:16 [Debug] failed to check availability of Xsc service:server response: 404 Not Found
404 page not found

Reporting to JFrog analytics is skipped...
16:21:16 [Info] Trace ID for JFrog Platform logs: e0d096f6bf504532
Error: 6 [Error] audit command in '/home/runner/_work/artifactory/artifactory/tests/e2e/maven' failed:
failed while building 'maven' dependency tree:
failed running command 'mvn org.apache.maven.plugins:maven-install-plugin:3.1.1:install-file -Dfile=/tmp/jfrog.cli.temp.-1723220476-1905448768/maven-dep-tree.jar -B': exec: "mvn": executable file not found in $PATH
```

**Comment by Fubuchi** [comment]
Running
```
 jf audit -h
```

Say that
```
--use-wrapper                    [Default: true] Set to false if you wish to not use the gradle or maven wrapper.
```
Which is a lie because our ci have the same issue as @coopernetes  jf audit doesn't use wrapper.

I even could not run `jf mvn-config --use-wrapper` on our ci because it run into interactive mode and asked me this question

![Image](https://github.com/user-attachments/assets/054af94f-5b69-455b-bbd7-ade82d08227f)
